spec/controllers/facilitator/projects_controller_spec.rb

```diff
@@ -3,64 +3,161 @@
 
+      context 'when params are valid' do
         let(:params) { { project_attributes: attributes_for(:project) } }
-        let(:project_attributes) { attributes_for(:project).merge(student_attributes) }
-        let(:student_attributes) { { student_attributes: attributes_for(:student) } }
 
-        it do
+        it 'redirects to root path and shows a success flash message on successful create' do
+          stringify_hash_values(params[:project_attributes])
           allow(ProjectCreationForm).to receive(:new).with(facilitator, params) { project_creation_form }
           allow(project_creation_form).to receive(:save) { true }
+          post :create, project_creation_form: params
           is_expected.to redirect_to(root_path)
           is_expected.to set_flash[:success]
         end
-
       end
 
       context 'when params are invalid' do
-
         let(:params) { { project_attributes: attributes_for(:project, :invalid) } }
 
-        it { is_expected.to render_template(:new) }
+        it 'renders new view on failed create' do
+          stringify_hash_values(params[:project_attributes])
+          allow(ProjectCreationForm).to receive(:new).with(facilitator, params) { project_creation_form }
+          allow(project_creation_form).to receive(:save) { false }
+          post :create, project_creation_form: params
+          is_expected.to render_template(:new)
+        end
+      end
+    end
+
+    describe '#edit' do
+      context 'facilitator access edit page' do
+        let(:project) { create(:project) }
+
+        before { get :edit, { id: project.id } }
+
+        it 'sets the project modification form with the selected project' do
+          expect(assigns(:project_modification_form).project).to eq(project)
+        end
 
+        it 'renders the edit page' do
+          is_expected.to render_template(:edit)
+        end
       end
+    end
 
+    describe '#update' do
+      let(:project_modification_form) { instance_double(ProjectModificationForm) }
 
+      context 'facilitator updates project' do
+        let(:project) { create(:project) }
+        let(:student_attributes) { { students_attributes: attributes_for(:student) } }
+        let(:project_attributes) { attributes_for(:project).merge(student_attributes) }
+
+        it 'redirects to root path and shows a success flash message' do
+          params = { project_attributes: {
+              name: "Renamed Project 1",
+              location: "Location X",
+              start_on: "2017-06-21",
+              end_on: "2017-10-18",
+              status: "1",
+              students_attributes: {
+                  name: "Student1",
+                  birthday: "2016-11-06",
+                  class_name: "1A",
+                  comments: nil
+              }}}
+
+          allow(ProjectModificationForm).to receive(:new).with(project, params) { project_modification_form }
+          allow(project_modification_form).to receive(:save) { true }
```

Comment:
Okay, have changed it. Still find this syntax a bit weird, since by stubbing it indirectly asserts as well, as it will fail if you don't present it the right params anyway.